src/Router/DataObject/Path.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare(strict_types=1);
+
+namespace YapepBase\Router\DataObject;
+
+use YapepBase\Exception\InvalidArgumentException;
+use YapepBase\Router\DataObject\Param\IParam;
+
+class Path
+{
+
+    /** @var string */
+    protected $pattern;
+
+    /** @var IParam[] */
+    protected $params = [];
+
+    /**
+     * @param string   $pattern
+     * @param IParam[] $params
+     */
+    public function __construct(string $pattern, array $params)
+    {
+        $this->pattern = $pattern;
+        $this->params  = $params;
+    }
+
+    /**
+     * @param $state array
+     *
+     * @return static
+     */
+    public static function __set_state($state): self
+    {
+        return new static(
+            $state['pattern'],
+            $state['params']
+        );
+    }
+
+    /**
+     * @param array $path
+     *
+     * @return Path
+     */
+    public static function createFromArray(array $path): self
+    {
+        if (!isset($path['pathPattern'])) {
+            throw new InvalidArgumentException('No path pattern set for path');
+        }
+
+        if (!is_string($path['pathPattern'])) {
+            throw new InvalidArgumentException(
+                'Invalid path pattern. Expected string, got ' . gettype($path['pathPattern'])
+            );
+        }
+
+        $pattern = '/' . trim($path['pathPattern'], "/ \r\n\t\0");
+        $params = [];
+
+        foreach ($path['params'] ?? [] as $index => $paramData) {
+            if (empty($paramData['type'])) {
+                throw new InvalidArgumentException('No type set for path ' . $pattern . ' param ' . $index);
+            }
+
+            $type = $paramData['type'];
+
+            if (isset(IParam::BUILT_IN_TYPE_MAP[$type])) {
+                $paramClass = IParam::BUILT_IN_TYPE_MAP[$type];
+            } else {
+                $paramClass = $type;
+            }
+
+            if (!class_exists($paramClass, true)) {
+                throw new InvalidArgumentException('Class ' . $paramClass . ' not found for path parameter');
+            }
+
+            if (!in_array(IParam::class, class_implements($paramClass, false))) {
```

Comment:
not instanceof, as the class is not yet instantiated, it's just a class name. The autoload is disabled because a few lines above we have already autoloaded it. Not important to disable it, but it's at least strange if we need to autoload it after a class_exists call on it.